### PR TITLE
Add administrate gem ...

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds the Administrate gem, and uses the built-in generator for dashboards for four existing resources:

* Detector::SuggestedResources
* SearchEvents
* Terms
* Users

Notable changes to the generator are:
1. The admin application generator gets a `require_user` method in addition to `authorize_user`, to ensure relevant error messages in all cases (confirmed with tests)
2. The Suggested Resources edit form has the fingerprint field removed, because that is a calculated field - so the user shouldn't have the opportunity to change it.

There is also a conditional link in the navigation, shown only to users who have the ability to access the default dashboard (currently set to the Terms area, but will change in a future ticket). This is a divergence from the pattern we're using in ETD, where the nav link is conditional on the admin flag on the User model (which I think is a pattern to avoid).

In hindsight, I think this PR would have been cleaner to have one commit of the generator output, and then a second commit to make the customization changes. If the reviewer would rather see this structure, let me know and I'll restructure accordingly.

### Ticket(s)

https://mitlibraries.atlassian.net/browse/TCO-51

### Accessibility

- [x] ANDI or Wave has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened 
      as new issues (link to those issues in the Pull Request details above)
- [ ] There are no accessibility implications to this change

### Documentation

- [ ] Project documentation has been updated
- [x] No documentation changes are needed

### ENV

- [ ] All new ENV is documented in README.
- [ ] All new ENV has been added to Heroku Pipeline, Staging and Prod.
- [x] ENV has not changed.

### Dependencies and migrations

YES dependencies are updated
NO migrations are included

### Code

- [ ] I have confirmed that the code works as intended.
- [ ] Any CodeClimate issues have been fixed or confirmed as
added technical debt.

### Documentation

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [ ] The documentation has been updated or is unnecessary.
- [ ] New dependencies are appropriate or there were no changes.

### Testing

- [ ] There are appropriate tests covering any new functionality.
- [ ] No additional test coverage is required.

### CODE REVIEW

File: app/controllers/admin/application_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    def require_user
+      return if current_user
+      
+      redirect_to root_path, alert: 'Please sign in to continue'
+    end
+
+    # Override this value to specify the number of elements to display at a time
+    # on index pages. Defaults to 20.
+    # def records_per_page
```
Comment (jazairi) [inline app/controllers/admin/application_controller.rb:32]: I assume this method was generated by the administrate install? I'm fine keeping it in; just want to make sure it wasn't a leftover comment from a WIP.

**Review by jazairi — APPROVED** [review]
Thanks for setting this all up! Everything looks right to me.

This also feels like a good place to share something I learned while researching the issue we were having yesterday with the 'lobby' view. It seems likely that the issue is not with administrate, but with using CanCanCan without a model.  If I'm correct about that, then we need to do two things to get authorization to work properly with the Stat/StatDashboard resource, [as suggested here](https://stackoverflow.com/a/69729278):

* In the controller, specify that there is no associated model: `authorize_resource class: false`
* In the Ability model, name the resource as a symbol: `can :read, :stat`

I tried the above yesterday and it didn't work for me. It was at the end of the day, though, so it might be worth digging into with fresh eyes. (I suspect I was doing something wrong with pluralization.)

File: app/controllers/admin/application_controller.rb (hunk 1 above)
Comment (matt-bernhardt) [reply app/controllers/admin/application_controller.rb:32]: Yeah, this was what the generator put in place - I remember noting it while working, and thinking "huh, I wonder if we'll ever actually use that".

**Comment by matt-bernhardt** [comment]
Ooh, interesting! Thanks for finding this. I've added it as a reference in the follow-up ticket: https://mitlibraries.atlassian.net/browse/TCO-63

**Pull request merged**